crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -28,21 +29,24 @@ use std::collections::HashSet;
 pub fn execute_call_entry_point(
     entry_point: &mut CallEntryPoint, // Instead of 'self'
     state: &mut dyn State,
-    cheatnet_state: &mut CheatnetState, // Added parameter
+    runtime_state: &mut RuntimeState,
     resources: &mut ExecutionResources,
     context: &mut EntryPointExecutionContext,
 ) -> EntryPointExecutionResult<CallInfo> {
     // region: Modified blockifier code
     // We skip recursion depth validation here.
 
-    cheatnet_state
+    runtime_state
+        .cheatnet_state
         .trace_data
-        .enter_nested_call(entry_point.clone());
+        .enter_nested_call(entry_point.clone()); // TODO this should be handled by trace extension
```

Comment:
There is an issue for that already #1488, tag it if you wanna leave a TODO

Reply:
Removed